sklearn/feature_extraction/hashing.py

```diff
@@ -113,11 +119,64 @@ def _validate_params(n_features, input_type):
             raise ValueError("input_type must be 'dict', 'pair' or 'string',"
                              " got %r." % input_type)
 
-    def fit(self, X=None, y=None):
-        """No-op.
+    def _transform(self, raw_X, save_mappings):
+        raw_X = iter(raw_X)
+        if self.input_type == "dict":
+            raw_X = (_iteritems(d) for d in raw_X)
+        elif self.input_type == "string":
+            raw_X = (((f, 1) for f in x) for x in raw_X)
+
+        if save_mappings:
```

Comment:
This is hard to read. Why not get all results in both cases then set it as an attribute conditionally?